src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.ts

```diff
@@ -0,0 +1,48 @@
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+import _ from 'lodash';
+
+import { Permission, Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+
+@Directive({
+  selector: '[cdScope]'
+})
+export class AuthStorageDirective {
+  permissions: Permissions;
+
+  constructor(
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef,
+    private authStorageService: AuthStorageService
+  ) {}
+
+  @Input('cdScope') set cdScope(condition: string | string[] | object) {
+    this.permissions = this.authStorageService.getPermissions();
+    if (this.isAuthorized(condition)) {
+      this.viewContainer.createEmbeddedView(this.templateRef);
+    } else {
+      this.viewContainer.clear();
+    }
+  }
+
+  @Input() cdScopeMatchAll = true;
+
+  private isAuthorized(
+    condition: string | string[] | Partial<{ [Property in keyof Permissions]: keyof Permission }>
+  ): boolean {
```

Comment:
Using a [type alias](https://www.typescriptlang.org/docs/handbook/declaration-files/by-example.html#reusable-types-type-aliases):
```ts
  type AdvancedCondition = Partial<{ [Property in keyof Permissions]: keyof Permission }>: boolean;
  
  @Input('cdScope') set cdScope(condition: string | string[] | AdvancedCondition) {
    ...
  }

  private isAuthorized(condition: string | string[] | AdvancedCondition) {
```

In fact, you could mix a type alias with a union of types:
```ts
  type Condition = string | string[] | Partial<{ [Property in keyof Permissions]: keyof Permission }>: boolean;
  
  @Input('cdScope') set cdScope(condition: Condition) {
    ...
  }

  private isAuthorized(condition: Condition)  {
```

Reply:
Thanks @epuertat